tutorials/graphql/intro-graphql/tutorial-site/content/graphql-vs-grpc.md

```diff
@@ -0,0 +1,88 @@
+
+## How does gRPC work?
+
+The "RPC" part of the name stands for "Remote Procedure Call", which indicates how the communication happens. In gRPC, the client communicates with the server by calling its methods as if they were local. That's possible by creating a stub (client) with the same methods as the gRPC server. You then use the client to call the server methods.
+
+They exchange data using Protobuf (Protocol buffers), compared to GraphQL, which uses text-based JSON to exchange data. Protobuf is more of a mechanism that serializes structured data into binary format. If you want to learn more about Protocol buffers, you can do it [here](https://developers.google.com/protocol-buffers/docs/overview).
+
+Similar to GraphQL, gRPC has a schema-like file, `.proto`, that describes the API service. It specifies the available methods that can be called and their parameters & return types. The default and most common Interface Definition Language (IDL) for writing "proto" files is Protobuf (Protocol buffers).
+
+Consider the following example:
+
+```
+syntax = "proto3";
+
+service User {
+  rpc GetUser (UserRequest) returns (UserResponse) {}
+}
+
+message UserRequest {
+  int32 id = 1;
+}
+
+message UserResponse {
+  string name = 1;
+  int32 age = 1;
+  string address = 1;
+}
+```
+
+The above code defines a service that allows you to retrieve the details about a specific user. The `GetUser` method represents the API endpoint the client can call to retrieve a user.
+
+Due to the `.proto` files and Protocol buffers, gRPC can auto-generate the client & server boilerplate code for yourself in various programming languages.
+
+## gRPC Benefits
+
+One benefit of using gRPC is the performance of data exchange. gRPC uses Protobuf to serialize data into binary format, which improves the payload size. As a result, exchanging data is faster and more efficient.
+
+gRPC also uses the HTTP/2 transport protocol, which comes with request and response multiplexing. In HTTP/2, the request/response is split into multiple frames that are sent individually and put together back when they arrive at the destination. That makes it possible to transfer multiple requests and responses in one connection.
+
+Another advantage of using gRPC is that it supports various streaming types. It supports
+* Server streaming - the client makes a request, and the server responds with a stream of messages
+* Client streaming - the client sends a stream of messages to the server, which responds after the client has finished streaming
+* Bidirectional streaming - both the client and server send independent streams of messages to each other
+
+gRPC also supports Unary interactions, where the client sends one request, and the server sends one response back.
+
+Code generation is another benefit. gRPC's `protoc` compiler can use the `.proto` file to generate server and client code in 11 programming languages.
```

Comment:
This is also possible via GraphQL codegen libraries. Can we add that note as well?

Reply:
The main idea of this section is that gRPC comes with this feature natively. GraphQL uses third-party tools.

I highlighted this in the document:
> gRPC comes with native support for code generation, whereas for GraphQL, you need third-party tools. gRPC can generate server and client code from the `.proto` file without requiring third-party tools.